NetworkingSampleApp/NetworkingSampleApp/Scenes/Authorization/AuthorizationViewModel.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  AuthorizationViewModel.swift
+//  NetworkingSampleApp
+//
+//  Created by Dominika Gajdová on 06.12.2022.
+//
+
+import Foundation
+import Networking
+import OSLog
+
+final class AuthorizationViewModel: ObservableObject {
+    private lazy var authManager = SampleAuthorizationManager()
+    private lazy var apiManager: APIManager = {
+        let authorizationInterceptor = AuthorizationTokenInterceptor(authorizationManager: authManager)
+        
+        var responseProcessors: [ResponseProcessing] = [
+            LoggingInterceptor.shared,
+            authorizationInterceptor,
+            StatusCodeProcessor.shared
+        ]
+        var errorProcessors: [ErrorProcessing] = [LoggingInterceptor.shared]
+        
+#if DEBUG
+        responseProcessors.append(EndpointRequestStorageProcessor.shared)
+        errorProcessors.append(EndpointRequestStorageProcessor.shared)
+#endif
+        
+        return APIManager(
```

Comment:
So basically every vm/store will create a new APIManager?

Reply:
If yes, I think APImanger init should have something like default interceptors which can be modified if needed vs creating them over and over

Reply:
Not really, we are going to have a shared APIManager.

Reply:
nope, we have one APIManager, only for auth you need its own, also for another REST API (if any third party or more servers) you may have various API manager setup

Reply:
I don't think we need a separate APIManager for auth requests, in case the `AuthorizationTokenInterceptor` is about to adapt or process a request that doesn't need authorization it simply ignores it. Not sure if @cejanen has something else in mind 🤔